src/api/firebase.js

```diff
@@ -162,6 +167,33 @@ export async function shareList(listPath, currentUserId, recipientEmail) {
 	}
 }
 
+/**
+ * Delete a list from a user's lists in Firestore.
+ * - If the deletion type is 'soft', the list will be removed from the user's view but not deleted from Firestore.
+ * - If the deletion type is 'hard', the list will be permanently deleted from Firestore and removed from the user's view.
+ *
+ * @param {string} deletionType - The type of deletion ('soft' or 'hard').
+ * @param {string} listPath - The path to the list document in Firestore.
+ * @param {string} userId - The id of the user who owns the list.
+ * @param {string} userEmail - The email of the user who has the list in their shared lists.
+ * @returns {Promise<string>} A message indicating the result of the deletion operation.
+ * @throws {Error} If something goes wrong during the deletion process.
+ */
```

Comment:
Thanks for adding documentation on this. Will be super helpful for anyone, but especially folks less familiar with firebase.